packages/hoppscotch-ui/src/components/smart/Table.vue

```diff
@@ -0,0 +1,129 @@
+                    {{ data }}
+                  </span>
+
+                  <span
+                    class="text-xs font-medium px-3 py-1 my-1 rounded-full bg-green-900 text-green-300 w-min"
+                  >
+                    {{ badgeName }}
+                  </span>
+                </div>
+              </div>
+
+              <!-- Column with Subtitles -->
+              <div
+                v-else-if="subtitleColumns.includes(colIndex.toString())"
+                class="flex flex-col truncate"
+              >
+                <span v-if="data" class="truncate">
+                  {{ data }}
+                </span>
+                <span v-else> - </span>
+
+                <div v-for="item in subtitles">
+                  <div
+                    v-if="item.colName === colIndex.toString()"
+                    class="text-gray-400 text-tiny"
+                  >
+                    <span>
+                      <span>
+                        {{ itemSubtitle(item.subtitle, rowIndex) }}
+                      </span>
+                    </span>
+                  </div>
+                </div>
+              </div>
+
+              <!-- Column with no subtitle or badge -->
+              <div v-else class="flex flex-col truncate">
+                <span v-if="data" class="truncate">
+                  {{ data }}
+                </span>
+                <span v-else> - </span>
+              </div>
+            </div>
+          </td>
+
+          <slot name="action" :item="item"></slot>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script lang="ts" setup generic="Item extends Record<string, unknown>">
+import { computed } from "vue"
+
+const props = defineProps<{
+  /** Whether to show the vertical border between columns */
+  yBorder?: boolean
+  /**  The list of items to be displayed in the table */
+  list: Item[]
+  /** The headings of the table */
+  headings: string[]
+  /** The styles to be applied to the table cells */
+  cellStyles?: string
+  /** The name of the badge */
+  badgeName?: string
+  /** The indices of the rows that needs to have a badge */
+  badgeRowIndices?: (number | undefined)[]
+  /** The index of the column that needs to have a badge */
+  badgeColName?: string
+  /** The subtitles to be displayed for the columns */
+  subtitles?: Array<{
+    /** The name of the column that needs to have a subtitle */
+    colName: string
+    /** The subtitle to be displayed for the column */
+    subtitle: string | string[]
+  }>
+}>()
+
+const emit = defineEmits<{
+  (event: "goToDetails", item: Item): void
+}>()
+
+// Returns all the columns that needs to have a subtitle
+const subtitleColumns = computed(() =>
+  props.subtitles ? props.subtitles.map((item) => item.colName) : []
+)
+
+// Returns the subtitle for the given column and row
+const itemSubtitle = (subtitle: string | string[], rowIndex: number) =>
+  typeof subtitle === "object" ? subtitle[rowIndex] : subtitle
```

Comment:
We should be able to access content from the parent component itself. We shouldn't have subtitles in the table comopnent. Make the table component generic. Just to render tr/td. We should be able to access row data from parent component like this `<template #row="item">`